lib/web_ui/lib/src/engine/browser_location.dart

```diff
@@ -96,9 +75,7 @@ class HashLocationStrategy extends LocationStrategy {
     // Dart will complain if a call to substring is
     // executed with a position value that exceeds the
     // length of string.
-    path = path.isEmpty ? path : path.substring(1);
-    // The path, by convention, should always contain a leading '/'.
-    return ensureLeading(path, '/');
+    return path.isEmpty ? path : path.substring(1);
```

Comment:
add assert(path[0] == '') before you substring ?

Reply:
I'm not sure I understand. `path` could either be an empty string, or a string that starts with `#`. Did you mean `assert(path[0] == '#')`?

Reply:
Ok, I added `assert(path.isEmpty || path.startsWith('#'));`